content/developer/reference/frontend/framework_overview.rst

```diff
@@ -383,5 +386,77 @@ It is mostly interesting for testing purposes: all code using the browser object
 can be tested easily by mocking the relevant functions for the duration of the
 test.
 
+.. _javascript/debug_mode:
+
+Debug mode
+==========
+
+Odoo can sometimes operate in a special mode called the `debug` mode. It is used
+for two main purposes:
+
+- display additional information/fields for some particular screens,
+- provide some additional tools to help developer debug the Odoo interface.
+
+The `debug` mode is described by a string. An empty string means that the `debug`
+mode is not active. Otherwise, it is active.  If the string contains `assets` or
+`tests`, then the corresponding specific sub modes are activated (see below). Both
+modes can be active at the same time, for example with the string `assets,tests`.
+
+The `debug` mode current value can be read in the :ref:`environment<javascript/environment>`:
+`env.debug`.
+
+.. tip::
+
+    To show menus, fields or view elements only in debug mode, you should target
+    the group `base.group_no_one`:
+    
+    .. code-block:: xml
+
+        <field name="fname" groups="base.group_no_one"/>
+
+Toggling debug mode
+-------------------
+
+The most common way to activate the debug mode is to edit the url: just add
+`?debug=1` or `?debug=assets` before the `#` sign, then reload the page. This
+will cause the server to mark the session as being in debug mode.
+
+To deactivate the debug mode, one can then simply add `?debug=0` to explicitely
+notify the server that we no longer want it.
+
+An alternative way to toggle the debug mode is to use the command palette. First
+step is to open it (ctrl+k), then type `debug`: a command will show up to
+activate the debug mode.
+
+Debug menu
+----------
+
+Once the debug mode is activated, an additional sub menu, called the `debug menu`,
+is available in the systray (in the top bar), or in the title bar of some
+dialogs. This menu contains additional tools that are useful to understand
+or edit technical data, such as the views or the actions. It contains
+some useful menu items such as:
+
+- edit action
+- manage filters
+- edit the current view
+- see the `fields view get`
+- and much more.
+
+Assets mode
+-----------
+
+The `debug=assets` sub mode is useful to debug javascript code: once activated,
+the :ref:`assets<reference/assets>` bundles are no longer minified, and source-maps
+are generated as well. This makes it useful to debug all kind of javascript code.
+
+Tests mode
+----------
+
+There is a specific sub mode named `tests`: if enabled, the server injects the
+bundle `web.assets_tests` in the page. This bundle contains mostly test tours
+(tours whose purpose is to test a feature, not to show something interesting to
+users). The `tests` mode is then useful to be able to run these tours.
```

Comment:
These sections are specific to the web client and should remain here, but a reference from the developer mode page would be neat.

There is actually already a tip mentioning the "assets" mode. Adding another one for the "tests" mode would do the job IMO.